qiskit/algorithms/eigen_solvers/numpy_eigen_solver.py

```diff
@@ -263,3 +322,64 @@ def compute_eigenvalues(
 
         logger.debug("EigensolverResult:\n%s", self._ret)
         return self._ret
+
+    def compute_transition_amplitudes(
+        self,
+        aux_operators: Optional[ListOrDict[OperatorBase]],
+        transition_amplitude_pairs: Optional[ListOrDict[Tuple[int, int]]],
+    ) -> Optional[ListOrDict[Tuple[complex, complex]]]:
+        """Computes the transition amplitudes for the desired auxiliary operators and eigenstates.
+        Args:
+            aux_operators: ListOrDict of auxiliary operators.
+            transition_amplitude_pairs: Dictionary specifying the names of the auxiliary operators
+            to consider, and the pairs of indices to take into account for the eigenstates.
+
+        Returns:
+            Dictionary of the resulting transition amplitudes. The keys of the result are
+            constructed with the following formula: aux_str + "_" + str(i) + "_" + str(j)
+
+        .. code-block::python
+
+            from qiskit.opflow import PauliSumOp
+            from qiskit.algorithms import NumPyEigensolver
+
+            qubit_op = PauliSumOp.from_list(
+                [
+                    ("II", -1.052373245772859),
+                    ("ZI", 0.39793742484318045),
+                    ("IZ", -0.39793742484318045),
+                    ("ZZ", -0.01128010425623538),
+                    ("XX", 0.18093119978423156),
+                ]
+            )
+            aux_op1 = PauliSumOp.from_list([("II", 2.0)])
+            aux_op2 = PauliSumOp.from_list([("II", 0.5), ("ZZ", 0.5), ("YY", 0.5), ("XY", -0.5)])
+            aux_ops = {"aux_op1": aux_op1, "aux_op2": aux_op2}
+            pairs = {"names": ["aux_op2"], "indices": [(0, 1), (1, 0), (1, 2)]}
+            algo = NumPyEigensolver(k=4)
+            algo.compute_eigenvalues(operator=qubit_op, aux_operators=aux_ops)
+            transition_amplitudes = algo.compute_transition_amplitudes(aux_ops, pairs)
+            # {'aux_op2_0_1': (0.0, 0.0),
+            #  'aux_op2_1_0': (0.0, 0.0),
+            #  'aux_op2_1_2': ((6.661338147750939e-16-0.7071067811865472j), 0.0)}
+        """
+
+        transition_amplitude_vals = None
+
+        if aux_operators is not None and transition_amplitude_pairs is not None:
+
+            restricted_aux_ops = {}
+            for name in transition_amplitude_pairs["names"]:
+                restricted_aux_ops[name] = aux_operators[name]
+
+            transition_amplitude_vals = {}
+            for pair in transition_amplitude_pairs["indices"]:
+                i, j = pair
+                temp_results = self._eval_transition_amplitudes(
+                    restricted_aux_ops, self._ret.eigenstates[i], self._ret.eigenstates[j]
```

Comment:
You should probably guard against `IndexError`s being raised here if either `i` or `j` exceed the number of available `eigenstates`

Reply:
Good point